commands/http/parse.go

```diff
@@ -20,12 +20,20 @@ func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
 
 	stringArgs := make([]string, 0)
 
+	if err := apiVersionMatches(r); err != nil {
```

Comment:
Been trying to make sense of this.
The client version header is set/hardcoded in the `commands` lib https://github.com/ipfs/go-ipfs/blob/9066b12d33cd67e083c15c58767c5a629c5165e2/commands/http/client.go#L104.
While the string `/go-ipfs/%s/` is ipfs specific, the api versioning scheme itself isn't.